perf/perf_lock.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2024 IBM
+# Author: Shaik Abdulla <abdulla1@linux.vnet.ibm.com>
+
+import os
+import platform
+from avocado import Test
+from avocado.utils import distro, process, dmesg
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class perf_lock(Test):
+
+    """
+    Tests perf lock and it's options namely record, report, info, script
+    contention with all possible flags with the help of yaml file
+    :avocado: tags=perf,lock
+    """
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.distro_name = detected_distro.name
+
+        deps = ['gcc', 'make']
+        if self.distro_name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(['perf'])
+        else:
+            self.cancel("Install the package for perf supported \
+                         by %s" % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        # Check perf lock is available in the system.
+        output = process.run('perf list | awk \'/lock:contention*/\'',
+                             shell=True).stdout.decode("utf-8")
```

Comment:
How hard to achieve this using python ?

Reply:
@PraveenPenguin we can achieve same output with this command 

        command = 'perf list | grep lock:contention*'
        output = process.run(command, shell=True).stdout.decode("utf-8")

Reply:
@FarooqAbdulla02 don't grep for lock:contention as its a subcommand in perf lock. we want to see if lock is enabled or not before running the test. You can use below command to check.
`perf list|grep -iw lock:.*`